src/cocotb_tools/pytest/plugin.py

```diff
@@ -0,0 +1,846 @@
+            level is inherited from the root logger. This behaves similarly to INFO.
+        """,
+    ),
+    Option(
+        "cocotb_log_prefix",
+        metavar="FORMAT",
+        help="""
+            Customize the log message prefix. The value of this variable should be in Python f-string syntax.
+            It has access to the following variables:
+            - record:  The LogRecord being formatted. This includes the attribute created_sim_time, which is the simulation time in steps.
+            - time:    The Python time module.
+            - simtime: The cocotb cocotb.simtime module.
+            - ANSI:    The cocotb cocotb.logging.ANSI enum, which contains ANSI escape codes for coloring the output.
+        """,
+    ),
+    Option(
+        "cocotb_pdb_on_exception",
+        action="store_true",
+        help="""
+            If enabled, cocotb will drop into the Python debugger (pdb) if a test fails with an exception.
+            See also the Python subsection of Attaching a Debugger.
+        """,
+    ),
+    Option(
+        "cocotb_plusargs",
+        nargs="*",
+        help="""
+            Plusargs are options that are starting with a plus (+) sign.  They are passed to the simulator and are
+            also available within cocotb as cocotb.plusargs. In the simulator, they can be read by the
+            Verilog/SystemVerilog system functions $test$plusargs and $value$plusargs.
+        """,
+    ),
+    Option(
+        "cocotb_reduced_log_fmt",
+        choices=("yes", "no"),
+        default="yes",
+        help="""
+            - yes: Logs will include simulation time, message type (INFO, WARNING, ERROR, ...), logger name, and the log message itself.
+            - no:  The filename and line number where a log function was called will be added between the logger name and the log message.
+        """,
+    ),
+    Option(
+        "cocotb_resolve_x",
+        choices=("error", "weak", "zeros", "ones", "random"),
+        help="""
+            Defines how to resolve bits with a value of X, Z, U, W, or - when being converted to integer. Valid settings are:
+            - error:  Resolves nothing.
+            - weak:   Resolves L to 0 and H to 1.
+            - zeros:  Like weak, but resolves all other non-0/1 values to 0.
+            - ones:   Like weak, but resolves all other non-0/1 values to 1.
+            - random: Like weak, but resolves all other non-0/1 values randomly to either 0 or 1.
+            There is also a slight difference in behavior of bool(logic).
+            When this is set, bool(logic) treats all non-0/1 values as equivalent to 0.
+            When this is not set, bool(logic) will fail on non-0/1 values.
+            Warning: Using this feature is not recommended.
+        """,
+    ),
+    Option(
+        "cocotb_results_file",
+        default=Path("results.xml"),
+        help="The file name where xUnit XML tests results are stored.",
+    ),
```

Comment:
Isn't this being overridden by the results collection process? We don't need to support this if pytest supports defining a particular junit output file.

Reply:
@ktbarrett I just blindly copied all `:envvar:` from cocotb documentation. Removed in this commit 4a6a69bd018de166cf2f04003180e0ec0865ed34